### PR TITLE
refactor(logger): Simplify sanitizeValue function

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Use `@sindresorhus/is` for type guards
- Get rid of implicit any
- Rewrite function in more linear manner
- Add test for non-plain objects handling

## Context:

- Ref: #12842

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/logger/utils.ts [hunk 1]
```diff
@@ -90,38 +91,54 @@ export default function prepareError(err: Error): Record<string, unknown> {
...
-    const arrayResult = Array(length);
-    seen.set(value, arrayResult);
-    for (let idx = 0; idx < length; idx += 1) {
-      const val = value[idx];
-      arrayResult[idx] = seen.has(val)
-        ? seen.get(val)
-        : sanitizeValue(val, seen);
-    }
-    return arrayResult;
+type NestedValue = unknown[] | object;
```
Comment (zharinov) [inline lib/logger/utils.ts:94]: It should be safer to use `object` instead of `Record<string, unknown>`, since we want sanitize class instances too.

**Review by rarkins — APPROVED** [review]

File: lib/logger/utils.ts [hunk 2]
```diff
@@ -90,38 +91,54 @@ export default function prepareError(err: Error): Record<string, unknown> {
...
-    if (value instanceof Date) {
-      return value;
+  if (is.array(value)) {
+    const length = value.length;
+    const arrayResult = Array(length);
+    seen.set(value, arrayResult);
+    for (let idx = 0; idx < length; idx += 1) {
+      const val = value[idx];
+      arrayResult[idx] =
+        isNested(val) && seen.has(val)
```
Comment (viceice) [inline lib/logger/utils.ts:132]: A string needs to be sanitized here too! So simply suggest to sanitize all val here

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/logger/utils.ts (hunk 2 above)
Comment (zharinov) [reply lib/logger/utils.ts:132]: It's done recursively via `sanitizeValue(val, seen)` for any non-nested value.  Any value from `seen` weak map  sanitized.

File: lib/logger/utils.ts [hunk 3]
```diff
@@ -90,38 +91,54 @@ export default function prepareError(err: Error): Record<string, unknown> {
...
+          ? seen.get(val)
+          : sanitizeValue(val, seen);
     }
+    return arrayResult;
+  }
 
+  if (is.object(value)) {
```
Comment (zharinov) [inline lib/logger/utils.ts:139]: This branch can be also improved with further PR, I think particular field checks should be refactored out

**Review by viceice — APPROVED** [review]

**Pull request merged**